### PR TITLE
Attempt at Nightly Coverity Scan and upload

### PR DESCRIPTION
**Pull request by bkuschel**
Signed-off-by: Boris Kuschel <Boris.Kuschel@ibm.com>

Changelog category (leave one):
- Build/Testing/Packaging Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add a night scan and upload for coverity.

> Information about CI checks: https://clickhouse.tech/docs/en/development/continuous-integration/

### CODE REVIEW

File: docker/packager/binary/Dockerfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # rebuild in #33610
-# docker build -t clickhouse/binary-builder .
+# COV_TOKEN=xxx docker buildx build --secret id=COV_TOKEN -t clickhouse/binary-builder .
```
Comment (Felixoid) [inline docker/packager/binary/Dockerfile:2]: We launch it from https://github.com/ClickHouse/ClickHouse/blob/master/tests/ci/docker_images_check.py#L236-L242, and I am afraid it requires much more adjustments

**Comment by bkuschel** [comment]
I changed it so the coverity tooling is fetched during execution. No need to rebuild image.

**Review by Felixoid — COMMENTED** [review]
A few moment to know

File: tests/ci/ci_config.py [hunk 2]
```diff
@@ -358,5 +369,8 @@
         "Performance Comparison (actions)": {
             "required_build": "performance",
         },
+        "Coverity Static Analysis (actions)": {
+            "required_build": "coverity",
+        },
```
[the remark this answers is not in the document]
Comment (bkuschel) [reply tests/ci/ci_config.py:374]: I wasn't sure, it can be removed.

**Review by Felixoid — COMMENTED** [review]
Besides the suggested changes, it's fine to merge. The secrets are needed to be adjusted from our side as well

**Pull request merged**